### DISCUSSION TITLE
Splitting text into separate columns from PDF

### DISCUSSION THREAD

**Discussion by PiotrKrosniak** [jsvine/pdfplumber#985] [Ask for help with specific PDFs]
I'm extracting tables from PDF document and my problem is that the script is splitting sentences into separate columns. How I can adjust table settings to force text to be in single column. 

```
p0 = pdf_nigeria.pages[46]
im = p0.to_image()
table_settings = {
    "vertical_strategy": "text",
    "horizontal_strategy": "text",
    "snap_y_tolerance": 6,
    
}
im.reset().debug_tablefinder(table_settings)
im
```
![image](https://github.com/jsvine/pdfplumber/assets/6203857/3e2fb451-d5bf-422e-869b-80074dc41544)

**Comment by samkit-jain** [comment]
Hi @PiotrKrosniak Appreciate your interest in the library. The "text" strategy for vertical lines does not always give you the desired results as it is more of a best guess implementation. You can use the explicit strategy to do the same. It can be done by using a table settings like
```python
{
    "vertical_strategy": "explicit",
    "explicit_vertical_lines": [100, 200, 300],  # These are the X coordinates where you want the vertical line separator to be present
    "horizontal_strategy": "text",
    "snap_y_tolerance": 6,   
}
```

You can learn more about them [here](https://github.com/jsvine/pdfplumber?tab=readme-ov-file#table-extraction-settings).
